python/snewpy/snowglobes_interface.py

```diff
@@ -153,26 +156,52 @@ def _load_smearing_matrices(self, path):
         logger.info(f'read smearing matrices for detectors: {list(self.smearings.keys())}')
         logger.debug(f'smearing matrices: {self.smearings}')
 
-    def _compute_rates(self, detector, material, flux_file:Path):
-        flux_file = flux_file.resolve()
-        fluxes = np.loadtxt(flux_file)
+    def compute_rates(self, detector:str, material:str, fluxes:np.ndarray, flux_energies:np.ndarray):
+        """ Calculate the rates for the given neutrino fluxes interacting in the given detector.
+
+        Parameters
+        ----------
+        detector: str
+            Detector name in SNOwGLoBES. Check `SimpleRate.detectors` for the list of options
+        material: str
+            Material name in SNOwGLoBES. Check `SimpleRate.materials` for the list of options
+        fluxes:
+            2d array of the neutrino fluxes.
+            First array dimension corresponds to flavors [nu_e, nu_mu, anti_nu_e, anti_nu_mu]
+            Second array dimension corresponds to the `energies` bins,
+
+        flux_energies:
+            1d array of the neutrino energy bins in GeV, corresponding to the `fluxes`
+
+        Returns
+        -------
+        pd.DataFrame
+            Table with Energy (GeV) as index values,
+            and number of events for each energy bin, for all interaction channels.
+            Columns are hierarchical: (is_weighted, channel),
+            so one can easily access the desired final table. 
+        """
         TargetMass = self.detectors[detector].tgt_mass
         data = {}
-        energies = np.linspace(7.49e-4, 9.975e-2, 200) # Use the same energy grid as SNOwGLoBES
-        if '_he' in detector:
-            energies = np.linspace(7.49e-4, 19.975e-2, 400) #SNOwGLoBES grid for he configurations
+
+        #load the binning for the smearing
+        binning= self.binning[material]
+        #calculate bin centers 
+        energies_t = 0.5*(binning['e_true'][1:]+ binning['e_true'][:-1] )
+        energies_s= 0.5*(binning['e_smear'][1:]+binning['e_smear'][:-1])
+        binsize = np.diff(binning['e_true'])
+
```

Comment:
In principle, truth and smeared energies should have the same binning, i.e. be the same (just # of events per energy bin will be different). In fact, if their binning is not equal, there should be an error. So, not sure we want to make the distinction.

Reply:
I just followed the format from (SNOw)GLoBES.

1. GLoBES uses distinct values for "energy" (e_smear) and "sample" (e_true) values, and that is what it expected for the smearing matrices, as I understood.
2. The code, generating the smearing matrices in SNOwGLoBES has these binning parameters separated:
https://github.com/SNOwGLoBES/snowglobes/blob/958cd2d891c873cd7e89addc1870d9bc940109f0/smearing_code/create_smearing_matrix.py#L24-L47

So I decided that we should conform to that, even if now `e_true` and `e_smear` are the same in all the smearing matrices, they might be different in the future. 

> In principle, truth and smeared energies should have the same binning

I think for some cases it would make sense them to be different. If `e_smear` is the detected energy, then for, say, neutron capture channels (like in HALO) we don't see the energy spectrum at all, just some monoenergetic lines. Then it might make sense to have smearing matrix for these channels to translate from full `e_true` range to several `e_smear` bins.
So I don't think we should raise an error if they are different - the code should work correctly for those.

Reply:
OK, indeed the matrix does not have to be squared (even if that's the case for the detectors implemented in snowglobes now) but to go from e_truth to e_smeared, the smearing_matrix needs to have same true energy binning as the e_truth to match the input for the multiplication, that's what I meant. 

I think you can keep the distinction as it is with no problem.

Reply:
Thanks! 

> smearing_matrix needs to have same true energy binning as the e_truth

Yes, I absolutely agree with that. As I understand the binning of the smearing matrix is what is written out in the `channels/channels_{material}` for some of the materials - and that's what we're reading in `_read_binning`.

If you agree on this implementation, could you please approve this PR?

Reply:
Yes!